### PR TITLE
Added Docker container support and documentation

### PR DESCRIPTION
**Pull request by paulyuk**
Implementing item in TODO list in Readme.developer.md.
This change adds Docker assets needed to build a single server container, and/or build a set of orchestrated containers for server and database together, configured by jsserver.properties with default host, port and admin, and with health checks to avoid timing issues.
Note we still need to remind the user to run create db and table scripts, however this is made easier by the scripts premounted in database container.  
Both quickstart commands and conceptual information have been added to a ## To develop and run with Docker containers and Docker Compose section of the Readme.developer.md documentation.
Without doing something unnatural, I did have to add Dockerfiles to root, and added supporting image content like property files and ODBC driver to a new /deploy folder (apologies I tried my best to tuck this away, but Docker files have a build context that want to see all included files in subdirectories, like \build\bin created by gradle). I'm open to alternate locations and styles if this is unwanted. 
In future updates it would be possible to make the entire build and test environment containerized too (imagine no need for gradle, python and other system installs), if that's interesting to the community.

### CODE REVIEW

**Comment by jdmonin** [comment]
Thank you, this looks interesting. I'll have time this weekend to look it over in detail.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM adoptopenjdk
```
Comment (jdmonin) [inline Dockerfile:1]: This should probably call for a specific version tag, for repeatability.
Example: adoptopenjdk:11-jre-hotspot

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+EXPOSE 8880
+ARG JAVA_OPTS
+ENV JAVA_OPTS=$JAVA_OPTS
+RUN mkdir /opt/app
+WORKDIR /opt/app
+COPY ./build/libs/JSettlersServer-2.3.00.jar /opt/app/japp.jar
```
Comment (jdmonin) [inline Dockerfile:7]: For `COPY` and `RUN`, i'd prefer to keep the jsettlers name instead of generic japp.jar.
/opt/app/JSettlersServer.jar for example

**Comment by jdmonin** [comment]
This mostly looks great, thanks again.

For several reasons, I don't want to have a copy of a JDBC jar in the repo. Instead the dockerfile could download it, for example:  
`RUN wget -O /opt/app/mariadb-java-client-2.6.0.jar https://downloads.mariadb.com/Connectors/java/connector-java-2.6.0/mariadb-java-client-2.6.0.jar`

I have a few comments on the Dockerfile and Readme.developer; see "Files changed" tab.

File: doc/Readme.developer.md [hunk 3]
```diff
@@ -424,6 +425,95 @@ jsettlers JAR from the command line, not running inside the IDE.
...
   `-Djsettlers.accounts.admins=adm`) which can then create others.
 
+## To develop and run with Docker containers and Docker Compose
+
+This is optional.  JSettlers Server and MariaDB (open source MySQL database) can be run 
+inside containers and honor properties set in `jsserver.properties`.  
+
+This dev-ops environment has the benefits that you can run the server and database
+in containers and eliminate all other dependencies in the runtime or development 
+environment (e.g. Java VM, JDK, MariaDb, ODBC drivers).  It also makes hosting
```
Comment (jdmonin) [inline doc/Readme.developer.md:435]: Do you mean JDBC (not ODBC)?

**Comment by paulyuk** [comment]
Thank you for review -- i haven't had time to properly maintain, so I'm closing out.

**Pull request closed without merging**